Documentation/Create SwiftSyntax Release.md

```diff
@@ -2,73 +2,10 @@
 
 ## Create release tag
 1. Check out the tag you want to release in the SwiftSyntax repository
```

Comment:
This section only contains a "checkout" step at this point, which doesn't really correspond to the title "Create release tag".